misc/gen-svg-props.hs

```diff
@@ -0,0 +1,355 @@
+#!/usr/bin/env stack
+-- stack --resolver lts-13.19 script
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE QuasiQuotes #-}
+
+-- | Regenerate the SVG attributes module.
+--
+-- Use from the top level of the repo, not the ./misc folder.
+--
+-- Eg:
+--
+--   $ ./misc/gen-svg-props.hs
+module Main where
+
+import Data.Foldable
+import Data.String.QQ
+import Data.Text (Text)
+import Data.Text.Encoding
+
+import qualified Data.ByteString as BS
+import qualified Data.ByteString.Lazy.Char8 as L8
+import qualified Data.Char as Char
+import qualified Data.Text as T
+
+
+main :: IO ()
+main =
+  BS.writeFile "./src/Concur/Replica/SVGProps.hs" (encodeUtf8 generatedFile)
+
+
+generatedFile :: Text
+generatedFile =
+  T.intercalate "\n" (fileStart : (uncurry attributeFunction <$> attributeMap))
+
+
+attributeFunction :: Text -> Text -> Text
+attributeFunction nameHs nameSvg =
+     nameHs <> " :: T.Text -> Props a\n"
+  <> nameHs <> " = textProp \"" <> nameSvg <> "\"\n"
+
+
+attributeMap :: [(Text, Text)]
+attributeMap =
+  let
+    svgNames :: [Text]
+    svgNames = filter (/= mempty) (T.splitOn "\n" attributeList)
+
+    camelCase :: Text -> Text
+    camelCase t =
+      let
+        f :: (Bool, Text) -> Char -> (Bool, Text)
+        f (capitalizeNext, acc) c
+          | c == ':' || c == '-' = (True, acc)
+          | capitalizeNext       = (False, T.snoc acc (Char.toUpper c))
+          | otherwise            = (False, T.snoc acc c)
+      in
+        snd (T.foldl' f (False, mempty) t)
+
+    toHaskellName :: Text -> Text
+    toHaskellName t =
+      case camelCase t of
+        "class" ->
+          "class'"
```

Comment:
Slight inconsistency here where DOM uses `className` and this uses `class'`. I could change this to `className`, but that would then make it inconsistent with the ones below.

Reply:
I propose we keep it in sync with `DOM.Props` here:

* `class` -> `className`
* `type` -> `type_` (see [here](https://github.com/pkamenarsky/concur-replica/blob/master/src/Concur/Replica/Props.hs#L324))
* `in` -> `in_`

Other than that, no complaints!

Reply:
I like it-- done.